### DISCUSSION TITLE
Stop Serenity Step from executing next steps when there is an error

### DISCUSSION THREAD

**Discussion by skhandekar1** [serenity-is/Serenity#5782] [Help]
I have a @Before method for setup, where I mark a setupComplete flag if setup completes successfully. but due to serenity executing all the steps even if there is an exception, the setupComplete flag gets set even when setup fails. what can I do to avoid this?

	@Before
	public void setUpTest() {	
			if (!setupComplete){
				user.userEntersNewCredentials();	
				user.userClickOnLogin();
				patientLocator.userShouldBeAbleToSearchPatientName("Thomas");
				patientLocator.userViewsPatientRecord();
				session.userShouldBeAbleToViewSessionListPageAfterClickingOnSessionTab();
				session.userShouldBeAbleToViewTheChooseSessionTypePopupAfterClickingOnPlusButton();
				session.userViewsCreateNewSessionPageAfterClickingOnORSessionButton();
				setAnodeCathode.setAnode("A1");
				setAnodeCathode.setCathode("A2");	
                                setupComplete = true;
			}			
	       }

**Comment by edwardch** [comment]
Hey @skhandekar1, welcome to the Serenity ASP.NET application framework repo. 

I think you might be in the wrong place, looks like you're trying to use [Serenity BDD](https://github.com/serenity-bdd/serenity-core).

Good luck with your testing!
